internal/acctest/datastore/ping_one_ldap_gateway_data_store_test.go

```diff
@@ -0,0 +1,225 @@
+
+type pingOneLdapGatewayDataStoreResourceModel struct {
+	dataStore *client.PingOneLdapGatewayDataStore
+}
+
+func initialPingOneLdapGatewayDataStore(pingOneConRef, pingOneEnvId, pingOneLdapGwId string) *client.PingOneLdapGatewayDataStore {
+	initialPingOneLdapGatewayDataStore := client.NewPingOneLdapGatewayDataStoreWithDefaults()
+	initialPingOneLdapGatewayDataStore.Id = pointers.String(pingOneLdapGatewayDataStoreId)
+	initialPingOneLdapGatewayDataStore.Name = pointers.String("initialPingOneLdapGatewayDataStore")
+	initialPingOneLdapGatewayDataStore.LdapType = ldapTypeVal
+	initialPingOneLdapGatewayDataStore.Type = pingOneLdapGDSType
+	initialPingOneLdapGatewayDataStore.PingOneConnectionRef = *client.NewResourceLink(pingOneConRef)
+	initialPingOneLdapGatewayDataStore.PingOneEnvironmentId = pingOneEnvId
+	initialPingOneLdapGatewayDataStore.PingOneLdapGatewayId = pingOneLdapGwId
+	return initialPingOneLdapGatewayDataStore
+}
+
+func updatedPingOneLdapGatewayDataStore(pingOneConRef, pingOneEnvId, pingOneLdapGwId string) *client.PingOneLdapGatewayDataStore {
+	updatedPingOneLdapGatewayDataStore := client.NewPingOneLdapGatewayDataStoreWithDefaults()
+	updatedPingOneLdapGatewayDataStore.Id = pointers.String(pingOneLdapGatewayDataStoreId)
+	updatedPingOneLdapGatewayDataStore.Name = pointers.String("updatedPingOneLdapGatewayDataStore")
+	updatedPingOneLdapGatewayDataStore.LdapType = ldapTypeVal
+	updatedPingOneLdapGatewayDataStore.Type = pingOneLdapGDSType
+	updatedPingOneLdapGatewayDataStore.PingOneConnectionRef = *client.NewResourceLink(pingOneConRef)
+	updatedPingOneLdapGatewayDataStore.PingOneEnvironmentId = pingOneEnvId
+	updatedPingOneLdapGatewayDataStore.PingOneLdapGatewayId = pingOneLdapGwId
+	updatedPingOneLdapGatewayDataStore.UseSsl = pointers.Bool(true)
+	updatedPingOneLdapGatewayDataStore.Name = pointers.String("myPingOneLdapGatewayDataStore")
+	updatedPingOneLdapGatewayDataStore.MaskAttributeValues = pointers.Bool(true)
+	updatedPingOneLdapGatewayDataStore.BinaryAttributes = []string{"binaryAttribute1", "binaryAttribute2"}
+	return updatedPingOneLdapGatewayDataStore
+}
+
+func TestAccPingOneLdapGatewayDataStore(t *testing.T) {
+	resourceName := "myPingOneLdapGatewayDataStore"
+
+	var pingOneConnectionRefId = os.Getenv("PF_TF_P1_CONNECTION_ID")
+	var pingOneEnvironmentId = os.Getenv("PF_TF_P1_CONNECTION_ENV_ID")
+	var pingOneLdapGatewayId = os.Getenv("PF_TF_P1_LDAP_GATEWAY_ID")
+
+	initialResourceModel := pingOneLdapGatewayDataStoreResourceModel{
+		dataStore: initialPingOneLdapGatewayDataStore(pingOneConnectionRefId, pingOneEnvironmentId, pingOneLdapGatewayId),
+	}
+
+	updatedResourceModel := pingOneLdapGatewayDataStoreResourceModel{
+		dataStore: updatedPingOneLdapGatewayDataStore(pingOneConnectionRefId, pingOneEnvironmentId, pingOneLdapGatewayId),
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			acctest.ConfigurationPreCheck(t)
+			var pingOneConnectionRefId = os.Getenv("PF_TF_P1_CONNECTION_ID")
```

Comment:
You ought to be able to use the above variables rather than re-calling Getenv here

Reply:
removed os.Getenv calls in latest commit